### ISSUE TITLE
Macros not emiting the line they are expanded on in nightly

### ISSUE THREAD

**Issue by enricostano** [rust-lang/rust#33280]
I noticed that running `try!()` macro in `main()` doesn't add line numbers in errors.

Code: http://is.gd/m7CFnx

It works fine in `stable` and `beta`, seems a `nightly` regression.

**Comment by jmesmon** [comment]
Might be related to #32747.

Appears that warnings/errors emitted by macros no longer are printing the line they were expanded on, and instead are just printing their line of definition twice.

Using this issue as an example:

Nightly gives:
```
<std macros>:5:8: 6:45 error: mismatched types:
 expected `()`,
    found `std::result::Result<_, _>`
(expected (),
    found enum `std::result::Result`) [E0308]
<std macros>:5 return $ crate :: result :: Result :: Err (
<std macros>:6 $ crate :: convert :: From :: from ( err ) ) } } )
<std macros>:5:8: 6:45 note: in this expansion of try! (defined in <std macros>)
<std macros>:5:8: 6:45 help: see the detailed explanation for E0308
error: aborting due to previous error
playpen: application terminated with error code 101
```

Beta/stable gives:
```
<std macros>:5:8: 6:42 error: mismatched types:
 expected `()`,
    found `core::result::Result<_, _>`
(expected (),
    found enum `core::result::Result`) [E0308]
<std macros>:5 return $ crate:: result:: Result:: Err (
<std macros>:6 $ crate:: convert:: From:: from ( err ) ) } } )
<anon>:2:5: 2:18 note: in this expansion of try! (defined in <std macros>)
<std macros>:5:8: 6:42 help: see the detailed explanation for E0308
error: aborting due to previous error
playpen: application terminated with error code 101
```

Differing lines:
```
-<anon>:2:5: 2:18 note: in this expansion of try! (defined in <std macros>)
+<std macros>:5:8: 6:45 note: in this expansion of try! (defined in <std macros>)
```

**Comment by jseyfried** [comment]
This was fixed in #33730.

**Comment by Mark-Simulacrum** [comment]
Closing, since this was fixed, it seems.
